base/compiler/typelattice.jl

```diff
@@ -239,6 +239,7 @@ function ⊑(@nospecialize(a), @nospecialize(b))
         return a === b
     end
 end
+⋤(@nospecialize(a), @nospecialize(b)) = !⊑(b, a)
```

Comment:
oh, right, we should be using `⊏` for the strict partial order, not `⋤`. So perhaps the "SQUARE IMAGE OF OR NOT EQUAL TO" is a reasonable choice of symbol here, which we use when we feel safe to assume that `a ⊑ b` is true.

Reply:
Any comment on 6bfb3a830c20d1cb2ca5e686a628ec85ef66ab6a ?

(Totally unnecessary and should be separated from this PR even if we do, but now I just started to feel an urge to rename `tmerge` to `⊔` and `tmeet` to `⊓` :p)

Reply:
LGTM. But would we also want to move towards doing:
```
at = at ⊓ fieldtype(t, i - 1)
if at ⋤ fieldtype(t, i - 1)
```
instead of
```
if at ⊏ fieldtype(t, i - 1)
```
as in the PR currently.

Reply:
16b1154d7e5f7a05adb0deeb0b58a67aea3b3cd7